File: backend/models/CompanyResearch.js
// models/CompanyResearch.js
const mongoose = require("mongoose");

const newsSchema = new mongoose.Schema({
  title: String,
  link: String,
  pubDate: String,
});

const companyResearchSchema = new mongoose.Schema({
  company: { type: String, required: true, unique: true },
  websiteUrl: { type: String },
  websiteText: { type: String }, // scraped content (optional to store)
  summary: { type: String }, // GPT summary
  news: [newsSchema], // all fetched news
  topNews: [newsSchema], // top 3 news
  companyDomain: { type: String }, // e.g. software, pharma, etc.
  employeeSize: { type: String }, // e.g. 100-500, 10k+, etc.
  companyLocations: [{ type: String }], // e.g. ["San Francisco, USA"]
  lastUpdated: { type: Date, default: Date.now },
});

module.exports = mongoose.model("CompanyResearch", companyResearchSchema);

File: backend/app/models/company_research.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class CompanyResearch(Base):
    __tablename__ = "company_research"

    id: Mapped[int] = mapped_column(primary_key=True)
    company: Mapped[str] = mapped_column(String(255), unique=True, index=True)
    official_website: Mapped[str] = mapped_column(String(500))
    website_text: Mapped[str] = mapped_column(Text)
    summary: Mapped[str] = mapped_column(Text)
    news: Mapped[list] = mapped_column(JSONB, default=list)
    top_news: Mapped[list] = mapped_column(JSONB, default=list)
    company_domain: Mapped[str] = mapped_column(String(255), default="")
    employee_size: Mapped[str] = mapped_column(String(255), default="")
    company_locations: Mapped[list] = mapped_column(JSONB, default=list)
    top_insights: Mapped[list] = mapped_column(JSONB, default=list)
    requested_by_user_id: Mapped[int | None] = mapped_column(ForeignKey("users.id"), nullable=True)
    last_updated: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, index=True)
